espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/ActivityHelpers.kt

```diff
@@ -84,9 +88,18 @@ object ActivityHelpers {
                     "with default options: $defaultOptions")
             makeIntent(defaultOptions)
         } else {
-            AndroidLogger.logger.info("Staring activity with custom options: $intentOptions")
-            makeIntent(intentOptions)
+            AndroidLogger.logger.info("Staring activity with custom options: ${params.optionalIntentArguments}")
+            makeIntent(params.optionalIntentArguments)
+        }
+
+        makeActivityOptions(params.optionalActivityArguments).let {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                instrumentation.startActivitySync(intent, it.toBundle())
+            } else {
+                instrumentation.targetContext.startActivity(intent, it.toBundle())
```

Comment:
I'm still not very convinced about direct `instrumentation.targetContext.startActivity` usage. The initial proposal was to only use it if `optionalActivityArguments` are set. If these are not set then the normal `instrumentation.startActivitySync(intent)` call should be used to preserve the backward compatibility

Reply:
Ok. Got it.